Upgrade xunit runner to 2.2.0-beta2-build3300

dagood opened the pull request:
This might solve https://github.com/dotnet/coreclr/issues/6716, stopping messages like these:

> One or more packages are incompatible with .NETFramework,Version=v4.5.

This also makes it unnecessary to generate and restore the project.json at runtime, because `xunit.runner.msbuild` can be moved into `tests/src/TestWrappersConfig/project.json` now that the incompatibility error doesn't happen. (Previously it happened, but was ignored.)

dagood commented:
/cc @ramarag @jasonwilliams200OK This might fix the "One or more packages are incompatible with .NETFramework,Version=v4.5." errors.

dagood commented:
@dsgouda This upgrades `xunit.runner.msbuild` used by `helix.targets`, does it look ok? 

(Linux ARM emulator error is low drive space: https://github.com/dotnet/coreclr/issues/6676)

dsgouda commented on tests/src/GC/Performance/Framework/GCPerfTestFramework.csproj:
```diff
@@ -59,12 +59,12 @@
+      <HintPath>packages\xunit.extensibility.execution.2.2.0-beta2-build3300\lib\net45\xunit.execution.desktop.dll</HintPath>
       <Private>True</Private>
     </Reference>
     <Reference Include="xunit.performance.core, Version=1.0.0.30, Culture=neutral, PublicKeyToken=67066efe964d3b03, processorArchitecture=MSIL">
```
Please upgrade xunit performance versions to 0040, also I doubt if we have a 0040 version for performance.metrics package you may want to remove that.

dagood replied:
Pushed a separate commit with the upgrade--I just upgraded both packages because they are available. There were a lot of dependency changes that VS also made as a result of those two upgrades.

dsgouda commented:
@dagood Given you are upgrading xunit-related dependencies I am pretty sure you will have to upgrade all the Microsoft.DotNet.xunit.performance.* packages to version 1.0.0-alpha-build0040 This will also need some testing using the Jenkins infra we now have to run perf tests.

dsgouda commented:
Tagging @smile21prc to run the perf tests via Jenkins and verify these changes won't affect the perf infra

dsgouda commented:
@dagood There are few other jsons that may still need the version change for xunit perf
https://github.com/dotnet/coreclr/search?l=json&q=xunit.performance&utf8=%E2%9C%93
Please make sure you have updated references under all the tests for consistency

dagood commented:
@dsgouda Thanks, upgraded them too.

dagood commented:
For this error:
```
09:45:37 D:\j\workspace\x64_release_w---0575cb46\Tools\publishtest.targets(124,5): error MSB4018: The "RemoveDuplicatesWithLastOneWinsPolicy" task failed unexpectedly. [D:\j\workspace\x64_release_w---0575cb46\bin\tests\Windows_NT.x64.Release\TestWrappers\baseservices\baseservices.XUnitWrapper.csproj]
09:45:37 D:\j\workspace\x64_release_w---0575cb46\Tools\publishtest.targets(124,5): error MSB4018: System.NullReferenceException: Object reference not set to an instance of an object. [D:\j\workspace\x64_release_w---0575cb46\bin\tests\Windows_NT.x64.Release\TestWrappers\baseservices\baseservices.XUnitWrapper.csproj]
09:45:37 D:\j\workspace\x64_release_w---0575cb46\Tools\publishtest.targets(124,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.RemoveDuplicatesWithLastOneWinsPolicy.Execute() [D:\j\workspace\x64_release_w---0575cb46\bin\tests\Windows_NT.x64.Release\TestWrappers\baseservices\baseservices.XUnitWrapper.csproj]
09:45:37 D:\j\workspace\x64_release_w---0575cb46\Tools\publishtest.targets(124,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [D:\j\workspace\x64_release_w---0575cb46\bin\tests\Windows_NT.x64.Release\TestWrappers\baseservices\baseservices.XUnitWrapper.csproj]
09:45:37 D:\j\workspace\x64_release_w---0575cb46\Tools\publishtest.targets(124,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext() [D:\j\workspace\x64_release_w---0575cb46\bin\tests\Windows_NT.x64.Release\TestWrappers\baseservices\baseservices.XUnitWrapper.csproj]
```
It seems to me that in [publishtest.targets#L127](https://github.com/dotnet/buildtools/blob/59bd1dbc2bee8dd54ba875a9cb7083dd3c13124d/src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets#L127), this would be caused by `_TestCopyLocalByFileName` containing no elements. Looking at what happens in a local build, I think [publishtest.targets#L64](https://github.com/dotnet/buildtools/blob/59bd1dbc2bee8dd54ba875a9cb7083dd3c13124d/src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets#L64) is the path taken, and that hard-coded buildtools version is wrong. Would it be correct to change the tests to use the `PrereleaseResolveNuGetPackageAssets` just above it by making sure the `ProjectLockJson` is accessible?

dagood commented:
@MattGal @rahku Do you have any advice on the issue in the above comment? (https://github.com/dotnet/coreclr/pull/6791#issuecomment-243838651)

I see now that helix.targets clears out the `ProjectLockJson`, and I don't know what the intention of that is. It seems like it blocks my idea of using a `ProjectLockJson` instead of the [hard-coded value in buildtools](https://github.com/dotnet/buildtools/blob/59bd1dbc2bee8dd54ba875a9cb7083dd3c13124d/src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets#L64).

MattGal commented:
@dagood, I'm not an expert on helix.targets (despite the name, noone on Helix wrote it).  I would say my advice is to get PrereleaseResolveNuGetPackageAssets  working, as it is a step towards executing more like CoreFX tests.  IIRC, @wtgodbe  is working on something like this.

rahku commented:
helix.targets maybe not be getting used anymore. You could remove the file and all references to it. @wtgodbe can you confirm, that you are not using the file(tests\helix.targets) anymore to enable helix integration with coreclr repo

wtgodbe commented:
I am not referencing helix.targets

On Aug 31, 2016, at 11:36 AM, Rahul Kumar <notifications@github.com<mailto:notifications@github.com>> wrote:


helix.targets maybe not be getting used anymore. You could remove the file and all references to it. @wtgodbe<https://github.com/wtgodbe> can you confirm, that you are not using the file(tests\helix.targets) anymore to enable helix integration with coreclr repo

-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/dotnet/coreclr/pull/6791#issuecomment-243856629>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ANnzeCGkQnx0McsnbNfMXr5gGHdqb2E1ks5qlcf7gaJpZM4Jno6D>.

dagood commented:
After removing helix.targets, my [changes to dir.props](https://github.com/dotnet/coreclr/pull/6791/files#diff-d5e4546d92c320a9ccf48fd6de2cf02d) got all-green CI so I've rebased to get rid of a merge conflict and squashed.

dagood commented:
@dotnet-bot test CentOS7.1 x64 Debug Build and Test (It hit https://github.com/dotnet/dotnet-ci/issues/307. The node is already marked for deletion.)

dagood commented:
Does this look good to merge, or are there some more tests anyone would like to do?

dsgouda commented:
I will leave it to your discretion but it still seems like some packages are quite old, eg. https://github.com/dotnet/corefx/blob/master/src/Common/test-runtime/project.json#L19 at the very least it would make sense to bring these packages up to speed with what we are using in CoreFX. 
P.S.: I am not sure if these changes will work with the Jenkins perf test runs @smile21prc is there a way to test these changes?

dagood commented:
@dsgouda Sure, I'll upgrade that to latest. I see in tests/scripts/project.json I downgraded it to make it even with everywhere else, which could have caused problems.

dsgouda commented:
In case upgrading in tests/scripts/project.json causes issues you can go ahead and merge these changes since I don't want to be a blocker but this is something you might want to consider moving forward

dagood commented:
It looks like the script project.json uses dotnet-buildtools, but the rest of the build only uses dotnet-core. Updated to latest in dotnet-core, although that's still 1.0.2, not the 1.0.3 that was in the scripts project.json.

dagood commented:
@dotnet-bot test Linux ARM Emulator Cross Debug Build

wtgodbe commented:
@smile21prc any reasons left not to merge this? Just want to make sure we're not breaking anything.

smile21prc commented:
As you mentioned this fix will fix the xunit.runner.msbuild 2.1.0 restore error in coreclr perf run, please go ahead check it in. Thanks.

14:32:54 RUNTEST: Invoking msbuild
14:33:51   
14:33:51   Errors in D:\j\workspace\perf_Windows_NTf52db88c\tests\\..\bin\tests\tempRestoreProject\project.json
14:33:51       Package xunit.runner.msbuild 2.1.0 is not compatible with net45 (.NETFramework,Version=v4.5). Package xunit.runner.msbuild 2.1.0 supports: portable-net45+win8+wp8+wpa81 (.NETPortable,Version=v0.0,Profile=Profile259)
14:33:51       One or more packages are incompatible with .NETFramework,Version=v4.5.
14:34:00   Desktop.Coreclr.TestWrapper -> D:\j\workspace\perf_Windows_NTf52db88c\bin\tests\Windows_NT.x64.Release\Common\Desktop.Coreclr.TestWrapper\Desktop.Coreclr.TestWrapper\Coreclr.TestWrapper.dll
14:34:00 D:\j\workspace\perf_Windows_NTf52db88c\tests\xunitwrapper.targets(9,5): warning : Your project.json doesn't list 'dotnet' as a targeted runtime. You should add '"dotnet": { }' inside your "runtimes" section in your project.json, and then re-run NuGet restore.
